File: math/3622-Check-Divisibility-by-Digit-Sum-and-Product/main.go
package main

func checkDivisibility(n int) bool {
	sumRes, mulRes := 0, 1
	nBackup := n

	for n > 0 {
		num := n % 10
		sumRes += num
		mulRes *= num

		n = n / 10
	}

	return nBackup%(sumRes+mulRes) == 0
}

File: math/3622-Check-Divisibility-by-Digit-Sum-and-Product/main.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        sum_num = 0
        mul_num = 1
        n_backup = n

        while n > 0:
            num = n % 10
            sum_num += num
            mul_num *= num
            n //= 10

        return n_backup % (sum_num + mul_num) == 0


# "/" and "//"
